### PR TITLE
#66-EB Create placeholder components for prompt/post page to connect frontend with API functions

### PR DESCRIPTION
**Pull request by eburdekin**
JIRA Ticket link: [PC-66](https://makeitmvp.atlassian.net/browse/PC-66)

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Refactoring

## Description

- **Changes made?** Added InputBox, PreviewBox, and Button components. Created PromptPage to hold those components. Need Scott and Mohan's PRs merged to add onClick functions to buttons, so added blank placeholder functions for now. Will add component styling later.
- **Reason for changes?** We need to create frontend components to generate OpenAI response, preview response, and post to LinkedIn.
Screen Shot 2024-04-17 at 1 23 51 PM

- **Behavior before?** N/A
- **Behavior after?** N/A
- **How did you test?** Previewed PromptPage - clicking Generate button inserts Prompt text into Preview box. Clicking edit button turns Preview box into editable textarea.  **NOTE:** since Scott's PR already adds react-router-dom to our package.json, I didn't redo that here, but we'll need to add PromptPage to the routes.
Screen Shot 2024-04-17 at 2 30 51 PM

## Checklist

- [x] I have performed a self-review of my code.
- [x] I have manually tested the new feature and found it to be working as expected.
- [x] All commits in this PR are related to the issue.
- [x] Base branch for this PR is set to `develop`.
- [x] JIRA ticket # and my initials are noted in this PR's name (format example: #57-EB Add Pull Request template)

**Reminder**: every PR needs at least 1 peer approval before merging to the `develop` branch. Once a main ticket has been completed, along with all subtasks, associated code can be merged to `main`.

[PC-66]: https://makeitmvp.atlassian.net/browse/PC-66?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by eburdekin** [comment]
@IguanasEverywhere @mohanhasti @shannon-n-lin ready for you guys to take a look. Very bare bones but the OpenAI call is hooked up to the frontend now.

File: src/pages/PromptPage/PromptPage.jsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  const [aiResponseLoading, setAiResponseLoading] = useState("");
+
+  const getOpenAIResponse = async () => {
+    setAiResponseLoading(true);
+    setPreviewText("Loading...");
+    requestObj.messages[1].content = inputText;
+    const responseContent = await getResponseContent(requestObj);
+
+    // need to parse JSON to return an object we can work with
+    const parsedContent = JSON.parse(responseContent.content);
```
Comment (IguanasEverywhere) [inline src/pages/PromptPage/PromptPage.jsx]: We may have an issue here if the response is an error. When I tested it with an error response, the app crashed. Can we change the order/check if we have response.content before parsing it and setting it to this variable?

File: src/pages/PromptPage/PromptPage.jsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  ],
+  model: "gpt-3.5-turbo",
+  temperature: 0.02,
+};
+
+const PromptPage = () => {
+  const [inputText, setInputText] = useState("");
+  const [previewText, setPreviewText] = useState("Preview here");
+  const [isEditing, setIsEditing] = useState(false);
+  const [aiResponseLoading, setAiResponseLoading] = useState("");
```
Comment (IguanasEverywhere) [inline src/pages/PromptPage/PromptPage.jsx]: I'm getting a warning about aiResponseLoading never being used (it starts as empty string, then gets set to true, and I don't think is used itself). Can we just use the previewText (or something similar)?

**Review by IguanasEverywhere — APPROVED** [review]
Looks awesome, Eileen! I just put a couple small questions in the PromptPage file, but personally I think this works great overall!

File: src/pages/PromptPage/PromptPage.jsx (hunk 1 above)
Comment (eburdekin) [reply src/pages/PromptPage/PromptPage.jsx]: Great catch, thank you @IguanasEverywhere ! Forgot to test again after adding JSON.parse()

**Pull request merged**